include/roaring/isadetection.h

```diff
@@ -207,14 +243,39 @@ static inline uint32_t croaring_detect_supported_architectures() {
 static inline bool croaring_avx2() {
   return false;
 }
+static inline bool croaring_avx512() {
+  return false;
+}
+#elif defined(__AVX512F__) && defined(__AVX512DQ__) && defined(__AVX512BW__) && defined(__AVX512VBMI2__) && defined(__AVX512BITALG__) && defined(__AVX512VPOPCNTDQ__)
+static inline bool croaring_avx2() {
+  return true;
+}
+static inline bool croaring_avx512() {
+  return true;
+}
 #elif defined(__AVX2__)
 static inline bool croaring_avx2() {
   return true;
 }
+static inline bool croaring_avx512() {
```

Comment:
I don't think that this is correct. If __AVX2__ is defined, it  could still be true that AVX-512 is supported.

Reply:
I agree, but if AVX-512 is supported. the upper AVX-512 branch will be compiled.